helix-core/src/main/java/org/apache/helix/monitoring/metrics/implementation/RebalanceLatencyGauge.java

```diff
@@ -30,6 +31,8 @@
   private static final Logger LOG = LoggerFactory.getLogger(RebalanceLatencyGauge.class);
   private static final long VALUE_NOT_SET = -1;
   private long _lastEmittedMetricValue = VALUE_NOT_SET;
+  // Use threadlocal here so the start time can be updated and recorded in multi-threads.
+  private final ThreadLocal<Long> _startTime;
```

Comment:
Wouldn't it be easier for each thread creating a new instance of RebalanceLatencyGauge, that you don't need to handle the concurrency complexities? It's more error-prone as we all know 
But I do see the usage pattern is the one single instance RebalanceLatencyGauge will be long-lived in the memory - if it's desired, can we make sure the class usage always follows the singleton pattern?

Reply:
It is necessary because we don't want multiple gauge objects. That will require us to assign different sensor names to those metrics. That would be very hard to manage.

Reply:
Thanks for the clarification. Is it possible to make the class a singleton pattern and limit the pubic constructor usages? @jiajunwang